cohortextractor/concepts/types.py

```diff
@@ -1,15 +1,34 @@
+from cohortextractor.sqlalchemy_types import TYPES_BY_NAME
+
+
 class BaseType:
-    pass
+    """
+    Base class for Column types defined in a Contract
+    """
+
+    # Subclasses must specify the backend Column types that they allow, as
+    # tuples of one or more keys from cohortextractor.sqlalchemy_types.TYPES_BY_NAME
+    allowed_backend_types: tuple[TYPES_BY_NAME, ...] = NotImplemented
```

Comment:
I think if you make this a subclass of Protocol then you can drop the ` = NotImplemented` and have the type system enforce it for you?

Reply:
brb off to read more about Protocol

Reply:
It worked 🎉

Reply:
Oops forgot to uncomment the line I used to test it :)